tests/rados/test_scrub_enhancement.py

```diff
@@ -0,0 +1,219 @@
+"""
+Program is for testing the scrub enhancement feature.As part of feature testing creating
+corrupted snapshot object.Performing scrub/deep-scrub to remove the corrupted SNA_ objects and
+replicating the correct SNA_ object
+"""
+
+import random
+import time
+import traceback
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.pool_workflows import PoolFunctions
+from ceph.rados.rados_scrub import RadosScrubber
+from tests.rados.rados_test_util import create_pools
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Verification of the scrub enhancement feature
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+    try:
+        log.info(run.__doc__)
+        global rados_obj, acting_osd_node, config, scrub_obj
+        config = kw["config"]
+        cephadm = CephAdmin(cluster=ceph_cluster, **config)
+        rados_obj = RadosOrchestrator(node=cephadm)
+        pool_obj = PoolFunctions(node=cephadm)
+        scrub_obj = RadosScrubber(node=cephadm)
+        client_node = ceph_cluster.get_nodes(role="client")[0]
+        mount_point = "~/automation_dir"
+        test_scenarios = ["scrub", "deep-scrub"]
+        log.info("Running scrub enhancement feature testing")
+        for test_input in test_scenarios:
+            log.info(f"The feature is testing withe the {test_input}")
+            log.info("Creating the pool with the single PG")
+            create_pools(config, rados_obj, client_node)
+            log.info("Pool creation completed")
+            oname = "UNIQUEOBJECT{i}".format(i=random.randint(0, 10000))
+            pool_name = config["create_pools"][0]["create_pool"]["pool_name"]
+            log.info("Setting the autoscale mode off")
+            rados_obj.set_pool_property(
+                pool=pool_name, props="pg_autoscale_mode", value="off"
```

Comment:
Suggestion : when pg_num_max is set to 1, then autoscaler mode to off is not required anymore. It can be removed.